services/management/annotation_test.go

```diff
@@ -0,0 +1,267 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package management
+
+import (
+	"context"
+	"testing"
+
+	"github.com/AlekSi/pointer"
+	"github.com/google/uuid"
+	"github.com/percona/pmm/api/managementpb"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"gopkg.in/reform.v1"
+	"gopkg.in/reform.v1/dialects/postgresql"
+
+	"github.com/percona/pmm-managed/models"
+	"github.com/percona/pmm-managed/utils/logger"
+	"github.com/percona/pmm-managed/utils/testdb"
+	"github.com/percona/pmm-managed/utils/tests"
+)
+
+func TestAnnotations(t *testing.T) {
+	setup := func(t *testing.T) (ctx context.Context, db *reform.DB, teardown func(t *testing.T)) {
+		t.Helper()
+
+		ctx = logger.Set(context.Background(), t.Name())
+		uuid.SetRand(new(tests.IDReader))
+
+		sqlDB := testdb.Open(t, models.SetupFixtures, nil)
+		db = reform.NewDB(sqlDB, postgresql.Dialect, reform.NewPrintfLogger(t.Logf))
+
+		teardown = func(t *testing.T) {
+			uuid.SetRand(nil)
+
+			require.NoError(t, sqlDB.Close())
+		}
+
+		return
+	}
+	authorization := "admin:admin"
+	authorizationHeaders := []string{authorization}
+
+	t.Run("Non-existing service", func(t *testing.T) {
```

Comment:
Let's add one more test with no services and no node in the request. To check that we add "pmm_annotations" tag.

Reply:
Here is test for it. Line 139. "Non-existing service and non-existing node".

Reply:
it's when we send request with non-existing service and node, but there might be request with empty services and node.

Reply:
Ok I will add it.